data/src/main/java/com/microsoft/azure/kusto/data/exceptions/TriState.java

```diff
@@ -0,0 +1,12 @@
+package com.microsoft.azure.kusto.data.exceptions;
```

Comment:
This class is quite generic. Do you intend to use this class elsewhere?

- On the one hand, this enum is only currently used for exception permanence. In that case, I think it would be clearer to name this enum "Permanence". It would also improve readability in new exception creation. Compare:
`throw new DataServiceException(clusterUrl, ERROR_ACQUIRING_APPLICATION_ACCESS_TOKEN, e, TriState.FALSE);`
vs
`throw new DataServiceException(clusterUrl, ERROR_ACQUIRING_APPLICATION_ACCESS_TOKEN, e, Permanence.FALSE);`
- On the other hand, if you envision using this in other cases (I personally don't), I suggest it should remain "TriState", but moved to a more generic package than "exceptions". I'd use "com.microsoft.azure.kusto.data.common" (which I think we should have anyway, for including anything that could be used by either the data or ingest projects).

- Lastly, an option I really like is to make use of existing Java functionality: Optional. Here's a sample usage:
`
public Optional<Boolean> isPermanent() {
    return Optional.of(true);
}
`

Reply:
will use Permanence
i started with an optional boolean but was asked to replace - i then tried to do something similar to the .Net but i guess we wont be using this elsewhere really